cocos/core/components/component.ts

```diff
@@ -362,15 +362,17 @@ class Component extends CCObject {
             // @ts-ignore
             const depend = this.node._getDependComponent(this);
             if (depend) {
-                return legacyCC.errorID(3626,
+                legacyCC.errorID(3626,
                     getClassName(this), getClassName(depend));
+                return false;
             }
         }
         if (super.destroy()) {
             if (this._enabled && this.node.activeInHierarchy) {
                 legacyCC.director._compScheduler.disableComp(this);
             }
```

Comment:
should return true here

Reply:
https://github.com/cocos-creator/engine/pull/6847/commits/0fa4c18192a2dbc7c13dd5b84f6f4887ff1af9c1